builds/Makefile

```diff
@@ -1429,48 +1429,53 @@ ifdef INCLUDE_DEVICE
 clean-local:: clean-llvm
 endif
 
-build-llvm32: .stamp-build-llvm .stamp-build-llvm32
-build-llvm64: .stamp-build-llvm .stamp-build-llvm32
+build-llvm32: .stamp-build-llvm32
+build-llvm64: .stamp-build-llvm64
 
-.stamp-build-llvm: $(SDK_CONFIG)
-	$(MAKE) -C $(SDK_BUILDDIR) provision-llvm-llvm64 $(SDK_ARGS)
+.stamp-compile-llvm64: $(SDK_CONFIG)
+	$(MAKE) -C $(MONO_SDK_BUILDDIR) provision-llvm-llvm64 $(SDK_ARGS)
 	$(Q) touch $@
 
-.stamp-build-llvm32: $(SDK_CONFIG)
-	$(MAKE) -C $(SDK_BUILDDIR) provision-llvm36-llvm32 $(SDK32_ARGS)
+.stamp-compile-llvm32: $(SDK_CONFIG)
+	$(MAKE) -C $(MONO_SDK_BUILDDIR) provision-llvm36-llvm32 $(SDK32_ARGS)
 	$(Q) touch $@
 
+.stamp-build-llvm64: .stamp-$(MONO_BUILD_MODE)-llvm64
+.stamp-build-llvm32: .stamp-$(MONO_BUILD_MODE)-llvm32
+
 clean-llvm: $(SDK_CONFIG)
-	$(MAKE) -C $(SDK_BUILDDIR) clean-llvm36-llvm32 clean-llvm-llvm64 $(SDK_ARGS)
+	$(MAKE) -C $(MONO_SDK_BUILDDIR) clean-llvm36-llvm32 clean-llvm-llvm64 $(SDK_ARGS)
 	$(RM) .stamp-*-llvm*
 
-.PHONY: install-llvm64 llvm llvm64
-
-install-llvm: install-llvm32 install-llvm64
+.PHONY: install-llvm64 llvm32 llvm64
 
-LLVM_TARGETS = \
+LLVM64_TARGETS = \
 	$(PREFIX)/LLVM/bin/opt \
 	$(PREFIX)/LLVM/bin/llc \
+
+LLVM32_TARGETS = \
 	$(PREFIX)/LLVM36/bin/opt \
-	$(PREFIX)/LLVM36/bin/llc
+	$(PREFIX)/LLVM36/bin/llc \
 
-$(PREFIX)/LLVM/bin/%: $(SDK_DESTDIR)/llvm-llvm64/bin/% | $(PREFIX)/LLVM/bin
-	$(call Q_2,INSTALL ,[LLVM64]) install -c -m 0755 $(INSTALL_STRIP_FLAG) $^ $@
+$(MONO_SDK_DESTDIR)/llvm-llvm64/bin/%: .stamp-build-llvm64; @true
```

Comment:
Why do you need `; @true`?

Reply:
This is a pattern rule, and pattern rules are a bit special.

What's happening here is that in `$(MONO_SDK_DESTDIR)/llvm-llvm64/bin/*` is made by the dependency `.stamp-build-llvm64`, and not by any recipe for pattern rule. This means that in theory I could have made this rule simpler, by not including a recipe.

The problem is that make doesn't consider pattern rules without recipes. So without the `@true`, make  complains that it doesn't know how to build anything that this rule is supposed to match.

BTW this is identical to:

```make
$(MONO_SDK_DESTDIR)/llvm-llvm64/bin/%: .stamp-build-llvm64
    @true
```

it's just in one line instead of two.